### PR TITLE
Fix local variable index of long and double arguments

### PR DESCRIPTION
**Pull request by saberduck**

### CODE REVIEW

File: java-frontend/src/test/java/org/sonar/java/bytecode/se/BytecodeEGWalkerTest.java [hunk 1]
```diff
@@ -172,6 +188,36 @@ public void test_method_throwing_exception() throws Exception {
     assertThat(methodYield).isInstanceOf(ExceptionalYield.class);
   }
 
+  @Test
+  public void test_starting_states() throws Exception {
+    BytecodeEGWalker walker = new BytecodeEGWalker(new BehaviorCache(null));
+    Symbol.MethodSymbol methodSymbol = mock(JavaSymbol.MethodJavaSymbol.class);
```
Comment (benzonico) [inline java-frontend/src/test/java/org/sonar/java/bytecode/se/BytecodeEGWalkerTest.java]: I think mock is unecessary here : new MethodJavaSymbol with new MethodJavaType should be sufficient and clearer.

File: java-frontend/src/main/java/org/sonar/java/bytecode/se/BytecodeEGWalker.java [hunk 2]
```diff
@@ -302,27 +303,33 @@ private void executeCheckEndOfExecution() {
...
-      state = state.put(i, sv);
+      state = state.put(parameterIdx, sv);
+      parameterIdx += operandStackSize(parameterType);
     }
     return Collections.singletonList(state);
   }
 
+  private static int operandStackSize(Type parameterType) {
+    // see JVM specification 4.10.2.3
+    return parameterType.isPrimitive(Type.Primitives.DOUBLE)
```
Comment (benzonico) [inline java-frontend/src/main/java/org/sonar/java/bytecode/se/BytecodeEGWalker.java]: weird formatting : keep the condition on one line.

File: java-frontend/src/test/java/org/sonar/java/bytecode/se/BytecodeEGWalkerTest.java [hunk 3]
```diff
@@ -172,6 +188,36 @@ public void test_method_throwing_exception() throws Exception {
...
+    JavaType intType = symbols.getPrimitiveFromDescriptor('I');
+
+    when(methodSymbol.parameterTypes()).thenReturn(ImmutableList.of(doubleType, intType));
+    startingState = Iterables.getOnlyElement(walker.startingStates(methodSymbol, ProgramState.EMPTY_STATE));
+    assertThat(startingState.getValue(0)).isNotNull();
+    assertThat(startingState.getValue(1)).isNull();
+    assertThat(startingState.getValue(2)).isNotNull();
+
+    // run symbolic execution of InnerClass.test_arg_indexes
+    getMethodBehavior(5);
```
Comment (benzonico) [inline java-frontend/src/test/java/org/sonar/java/bytecode/se/BytecodeEGWalkerTest.java]: No assertion ?

Comment (saberduck) [inline java-frontend/src/test/java/org/sonar/java/bytecode/se/BytecodeEGWalkerTest.java]: I decided to drop this indirect test in the end. It was redundant

**Pull request merged**